src/coreclr/gc/gcee.cpp

```diff
@@ -474,6 +474,15 @@ segment_handle GCHeap::RegisterFrozenSegment(segment_info *pseginfo)
 #endif // FEATURE_BASICFREEZE
 }
 
+void GCHeap::UpdateFrozenSegment(segment_handle seg, uint8_t* allocated, uint8_t* commited)
+{
+#ifdef FEATURE_BASICFREEZE
+    heap_segment* heap_seg = reinterpret_cast<heap_segment*>(seg);
+    heap_segment_allocated (heap_seg) = allocated;
```

Comment:
There can be background GC running when this is done. Is background GC robust against this changing without any locks?

Reply:
Good question, I am not sure 🙁

Reply:
@Maoni0 Is it ok to modify heap_segment_allocated and commited like this?

Reply:
And is it ok that I bump it before I update object's methodtable

Reply:
that's fine because for BGC we only operate between the range of background_saved_lowest_address and background_saved_highest_address, which would be guaranteed to be committed. "operate" means we touch various GC data structures. I have not looked at  the rest of this PR but I presume, based on this question, that you will not ever do this during a blocking GC. so if that's true then it's fine.